devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp

```diff
@@ -605,10 +626,42 @@ bool YarpRobotLoggerDevice::setupRobotCameraBridge(
     return true;
 }
 
+bool YarpRobotLoggerDevice::initMetadata(const std::string& nameKey, const std::vector<std::string>& metadataNames)
+{
+    bool ok = m_bufferManager.addChannel({nameKey, {metadataNames.size(), 1}, metadataNames});
+    std::string rtNameKey = robotRtRootName + treeDelim + nameKey;
+    if (m_sendDataRT)
+    {
+        ok &= m_vectorCollectionRTDataServer.populateMetadata(rtNameKey, metadataNames);
+    }
+    return ok;
+}
+
+void YarpRobotLoggerDevice::logData(const std::string& name,
+                                                    const Eigen::VectorXd& data,
+                                                    const double time)
+{
+    m_bufferManager.push_back(data, time, name);
+    std::string rtName = robotRtRootName + treeDelim + name;
+    if(m_sendDataRT)
+    {
+        m_vectorCollectionRTDataServer.populateData(rtName, data);
+    }
```

Comment:
Can you directly expose the name instead of `rtName`?

Reply:
Can you clarify? What do you mean by directly expose?